libraries/disp/viewers/averageselectionview.cpp

```diff
@@ -98,114 +147,70 @@ void AverageSelectionView::init()
         delete child;
     }
 
-    //Set trigger types
-    QMapIterator<double, QPair<QColor, QPair<QString,bool> > > i(m_qMapAverageInfo);
+    // Create new GUI elements
+    QMapIterator<QString, QColor> itr(*m_qMapAverageColor);
     int count = 0;
-    m_qMapButtonAverageType.clear();
-    m_qMapChkBoxAverageType.clear();
+    while(itr.hasNext()) {
+        if(count >= m_iMaxNumAverages) {
+            break;
+        }
 
-    while (i.hasNext()) {
-        i.next();
+        itr.next();
 
-        //Create average checkbox
-        QCheckBox* pCheckBox = new QCheckBox(i.value().second.first);
-        pCheckBox->setChecked(i.value().second.second);
+        //Create average active checkbox
+        QPointer<QCheckBox> pCheckBox = new QCheckBox(itr.key());
+        pCheckBox->setChecked(m_qMapAverageActivation->value(itr.key()));
+        pCheckBox->setObjectName(itr.key());
         topLayout->addWidget(pCheckBox, count, 0);
-        connect(pCheckBox, &QCheckBox::clicked,
+        connect(pCheckBox.data(), &QCheckBox::clicked,
                 this, &AverageSelectionView::onAveragesChanged);
-        m_qMapChkBoxAverageType.insert(pCheckBox, i.value().second.first.toDouble());
 
         //Create average color pushbutton
-        QPushButton* pButton = new QPushButton();
-        pButton->setStyleSheet(QString("background-color: rgb(%1, %2, %3);").arg(i.value().first.red()).arg(i.value().first.green()).arg(i.value().first.blue()));
+        QColor color = itr.value();
+        QPointer<QPushButton> pButton = new QPushButton("Click to change");
+        pButton->setObjectName(itr.key());
+        pButton->setStyleSheet(QString("background-color: rgb(%1, %2, %3);").arg(color.red()).arg(color.green()).arg(color.blue()));
         topLayout->addWidget(pButton, count, 1);
-        connect(pButton, &QPushButton::clicked,
+        connect(pButton.data(), &QPushButton::clicked,
                 this, &AverageSelectionView::onAveragesChanged);
-        m_qMapButtonAverageType.insert(pButton, i.value().second.first.toDouble());
 
-        ++count;
+        count++;
```

Comment:
++count is one cpu step shorter, i.e., faster - pre-increment instead of post-increment
```suggestion
        ++count;
```